src/writer.rs

```diff
@@ -427,6 +428,33 @@ where
             .write_event(Event::End(BytesEnd::borrowed(b"ListStyle")))?)
     }
 
+    fn write_link_type(&mut self, link: &LinkType) -> Result<(), Error> {
+        let (tag_name, link_model) = match link {
+            LinkType::Icon(model) => (b"Icon", &*model),
+            LinkType::Link(model) => (b"Link", &*model),
+        };
+        self.writer
+            .write_event(Event::Start(BytesStart::owned_name(tag_name.to_vec())))?;
```

Comment:
We'll need to add this here to capture the attributes

```rust
.with_attributes(self.hash_map_as_attrs(&link_model.attrs))
```

Reply:
Done.